Fix primitive ws (#880)
facebookresearch/hydra: website/docs/advanced/override_grammar/basic.md

## Grammar
Hydra supports a rich [DSL](https://en.wikipedia.org/wiki/Domain-specific_language) in the command line.   
Below are the parser rules from grammar.
You can see the full grammar on GitHub

> ([lexer](https://github.com/facebookresearch/hydra/tree/master/hydra/grammar/OverrideLexer.g4) and

Reviewer: I think I didn't answer your question:

Yes, please add a paragraph about whitespace being supported in values and a mention that it has to be quoted for the shell.

Author: I added a sub-section on unquoted strings with shell examples in 6434fc6206726c6a51494e0607e0546c387eeeda, let me know if that looks good to you.

[parser](https://github.com/facebookresearch/hydra/tree/master/hydra/grammar/OverrideParser.g4))

> ```antlr4

Reviewer: ```suggestion
```antlr4 title="OverrideParser.g4"
```

Author: Committing your suggestion

    | function
;

argName: ID EQUAL;

> function: ID PARENTHESIS_OPEN

Reviewer: POPEN, PCLOSE. this is too long :)

Author: Done in 047b7c23

</div>
</div>

### Unquoted values
Override values can contain whitespaces and other special characters.
Some examples of legal values: `hello world`,`   hello world  `, `https://${server}:${env:PORT}`.
Unquoted whitespaces are allowed in the middle of the word, while leading and trailing whitespaces are stripped.
Normally, your shell will interpret values with whitespaces as being multiple parameters (`key=a b` would be interpreted as `key=a` and `b`).
To prevent this you can quote them with a single quote. For example:

```bash
$ python my_app.py 'msg=hello world'
```

> 

Reviewer: What do you think about (not suggestion because GitHub does not render it correctly).

### Whitespaces in unquoted values
Unquoted Override values can contain non leading or trailing whitespaces.
For example, `msg=hello world` is a legal override (key is `msg` and value is the string `hello world`).
Normally, your shell will interpret values with whitespaces as being multiple parameters (`key=a b` would be interpreted as `key=a` and `b`).
To prevent this you can quote them with a single quote. For example:

```bash
$ python my_app.py 'msg=hello world'
```

Note that trailing and leading whitespace are ignored, the above is equivalent to:
```bash
$ python my_app.py 'msg=    hello world    '
```

Author: Looks good to me, I pushed it in b54615cc

</div>
</div>

> ### Unquoted strings

Reviewer: ```suggestion
### Unquoted values
```

Author: Taking your suggested commit

> Hydra also allows unquoted strings to simplify the command line syntax.
> These strings can be made of alphanumeric characters, as well as the following symbols: `_/\+-.:$*`.
> Whitespaces are also allowed in-between these characters (but leading and trailing whitespaces are stripped).
> Finally, interpolations may be embedded within an unquoted string.
> 
> Here are some examples showing how these may be used on the command line (note that here all
> quotes are "consumed" by the shell, so that from Hydra's point of view all these strings are unquoted):
> ```bash
> python my_app.py url=https://hydra.cc
> python my_app.py url='https://${server}:${env:PORT}'
> python my_app.py msg='hello world'
> python my_app.py msg='  hello world  '  # same as above
> python my_app.py msg='hello^world'      # invalid, use a quoted '"hello^world"' instead

Reviewer: I would prefer to train people to quote the whole override when quoting for the shell.
```suggestion
Override values can contain whitespaces and other special characters.
Some examples of legal values: `hello world`,`   hello world  `, `https://${server}:${env:PORT}`.
Unquoted whitespaces are allowed in the middle of the word, while leading and trailing whitespaces are stripped.
Normally, your shell will interpret values with whitesspaces as being multiple parameters (`key=a b` would be interpreted as `key=a` and `b`). To prevent this you can quote them with a single quote. For example:

```bash
$ python my_app.py 'msg=hello world'
```

Author: Taking your suggested commit -- will just follow-up with a small typo fix in it
